sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -160,11 +162,15 @@ case class Abs(child: Expression, failOnError: Boolean = SQLConf.get.ansiEnabled
 
   def this(child: Expression) = this(child, SQLConf.get.ansiEnabled)
 
-  override def inputTypes: Seq[AbstractDataType] = Seq(NumericType)
+  override def inputTypes: Seq[AbstractDataType] = Seq(TypeCollection.NumericAndInterval)
```

Comment:
We shouldn't use `TypeCollection.NumericAndInterval`, as it includes the legacy interval. We will hit runtime exception if we use legacy interval as input.

Reply:
@cloud-fan Thanks for the comment. Here is the fix https://github.com/apache/spark/pull/34183